### PR TITLE
feat: Implement testcase for PATCH basic info

### LINKED ISSUE #644: Implement testcase for device profile PATCH basic info
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
 This feature request is for metadata.

### Description [**REQUIRED**]
According to https://docs.edgexfoundry.org/2.2/design/adr/core/0021-Device-Profile-Changes/ & https://github.com/edgexfoundry/edgex-go/issues/3925
The PATCH device profile basic info implement tests would be added to metadata.

### Describe the solution you'd like
 If you have a solution in mind, please describe it.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by jumpingliu**
fix #644

Signed-off-by: melody <melody@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

**Comment by jumpingliu** [comment]
Since keyword` Generate basicinfo ` is used in the positive and negative cases, I put it in the coreMetadataAPI file.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-Basicinfo.robot [hunk 1]
```diff
@@ -4,26 +4,30 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
-ErrProfileBasicInfoPATCH002 - Update basicinfo with invalid profile name
-    #Non-existent profile name
-    Given Generate A Device Profile Sample for basicinfo  Test-Profile-5
-    And Set To Dictionary  ${deviceProfile}[0][profile]  name=Non-existent
-    When Update Basicinfo ${deviceProfile}
-    And Should Return Status Code "404"
-    [Teardown]  Delete Device Profile By Name
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete device profile by name  Test-Profile-1
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-Basicinfo.robot:22]: You have set`Test-Profile-1` to `test_profile` test variable in the `Generate a basicinfo sample for updating`, so use `test_profile` here instead.
```suggestion
    [Teardown]  Delete device profile by name  ${test_profile}
```

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-Basicinfo.robot [hunk 2]
```diff
@@ -4,26 +4,30 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
+ErrProfileBasicInfoPATCH002 - Update basicinfo with Non-existent profile name
+    # Non-existent profile name
+    Given Generate a basicinfo sample for updating
+    And Set To Dictionary  ${basicinfoProfile}[0][basicinfo]  name=Non-existent
+    When Update basicinfo ${basicinfoProfile}
+    Then Should Return Status Code "207"
+    And Item Index 0 Should Contain Status Code "404"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete device profile by name  Test-Profile-1
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-Basicinfo.robot:33]:
```suggestion
    [Teardown]  Delete device profile by name  ${test_profile}
```

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**